File: Java/Node2D.java
public class Node2D {

    private Node2D nextNode;
    private LinkedList list;

    Node2D(){
        this.list = new LinkedList();
    }

    Node2D(LinkedList list){
        this.list = list;
    }

    public Node2D getNextNode(){
        return this.nextNode;
    }

    public void setNextNode(Node2D nextNode){
        this.nextNode = nextNode;
    }
    
    /* populates Node2D with list of given size */
    public void populateList(int size){
        Random rand = new Random();
        int tempInt;
        for(int i = 0; i < size; i++){
            tempInt = rand.nextInt(2147483647);   
            Node temp = new Node(tempInt);
            list.insertAtBeginning(temp);
        }
    }

    /* populates Node2D with cycle of given size */
    public void populateCycle(int size){
        Random rand = new Random();
        int tempInt;
        for(int i = 0; i < size-1; i++){
            tempInt = rand.nextInt(2147483647);   
            Node temp = new Node(tempInt);
            list.insertAtBeginning(temp);
        }
        Node temp = new Node(516);
        list.insertCycle(temp);
    }

}

File: Python/Node2D.py
import random
from LinkedList import LinkedList
from Node import Node

class Node2D(object):

    def __init__(self, nextnode=None):
        self.nextNode = nextnode
        self.list = LinkedList.initialize()

    @classmethod
    def initialize(cls):
        ll = cls()
        return ll

    @classmethod
    def initializewithargs(cls, somelist):
        ll = cls()
        ll.list = somelist
        return ll

    def getnextnode(self):
        return self.nextNode

    def setnextnode(self, nextnode):
        self.nextNode = nextnode

    def populatelist(self, size):
        for i in range(size):
            tempint = random.randrange(1, 2147483647)
            tempnode = Node.initializewithargs(tempint)
            self.list.insertatbeginning(tempnode)

    def populatecycle(self, size):
        for i in range(size):
            tempint = random.randrange(1, 2147483647)
            tempnode = Node.initializewithargs(tempint)
            self.list.insertatbeginning(tempnode)
        tempnode = Node.initializewithargs(516)
        self.list.insertcycle(tempnode)


